### DISCUSSION TITLE
Live Update: bundle is preserved after app update?

### DISCUSSION THREAD

**Discussion by mustafa0x** [capawesome-team/capacitor-plugins#246] [Q&A] [answered]
I am testing the live update plugin locally, on a simulator. I updated the bundle (`downloadBundle`, `setBundle`, `reload`). Then I made some changes to my app and ran `npm exec cap run`. When the new app version opened on the simulator, the old bundle was active!

My assumption is that live update bundles are wiped if the app (ie apk/aab/ipa) changes. My concern is that this behavior will also occur when the user updates the app.

Edit,

- I could compare the active `bundleId` with `LiveUpdate.getVersionName()`, and call `deleteBundle` if they're out of match, but my concern is the app might not even start, if the native code has changed drastically.
- I ran `adb install -r android/app/build/outputs/apk/debug/app-debug.apk` and indeed, the live update bundle was preserved. Even though the `versionName` changed (but `versionCode` didn't).

**Comment by mustafa0x** [comment]
Update: it seems to use the new bundle, but `(await LiveUpdate.getBundles()).bundleIds.at(-1)` returns the old version on first load. I'm still investigating. I would also appreciate any advice.

**Comment by robingenz** [comment, marked as answer]
Hey, sorry for my late answer. For some reason I did not receive a notification for this discussion.

> I am testing the live update plugin locally, on a simulator. I updated the bundle (`downloadBundle`, `setBundle`, `reload`). Then I made some changes to my app and ran `npm exec cap run`. When the new app version opened on the simulator, the old bundle was active!

Yes, this is because the app no longer uses the default bundle but the latest live update. If you run `npm exec cap run`, the default bundle will be overwritten, but it is not currently in use because you have set another bundle via the live update plugin. You then have three options during development:
1. Call the [`reset()`](https://capawesome.io/plugins/live-update/#reset) method to reset the app to the default bundle
2. Reinstall the app so that all data is deleted
3. Increment the `versionCode` / `CFBundleVersion` so that the plugin automatically performs a reset

However, this is only a problem during development. It is NOT a problem in production, as the plugin automatically performs a reset during a native update, as the `versionCode` / `CFBundleVersion` is always incremented during a native update.

**EDIT**: I have just added this information to the docs: https://github.com/capawesome-team/capacitor-plugins/commit/9b1c27d0858262052772e6f34e182709b27a3143

**Reply by mustafa0x** [reply]
> However, this is only a problem during development. It is NOT a problem in production, as the plugin automatically performs a reset during a native update, as the `versionCode` / `CFBundleVersion` is always incremented during a native update.

Awesome, this is exactly the confirmation I needed. And now that I reviewed the docs I noticed the `resetOnUpdate` option. Thanks!
